### PR TITLE
In an ansible collection playbooks are in playbooks directory

### LINKED ISSUE #514: FQCN playbooks do not work on run_playbook action
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/ansible-rulebook/issues) for duplicates.
- [X] I understand that ansible-rulebook is open source software provided for free and that I might not receive a timely response.

### Bug Summary

Using FQCN playbooks on run_playbook they don't seem to work.
The rulebook process errors out that it cannot find it.

### Environment

Fedora 34

### Steps to reproduce

1. Clone ansible/aap-containerized-installer repo
2. Install it as a collection by running `ansible-galaxy collection install .'
3. Create a rulebook with a rule with action such as:

```
      action:
        run_playbook:
          name: ansible.containerized_installer.install
```

### Actual results

When the event gets triggered, I get:

```
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]: 2023-05-24 12:43:24,546 - ansible_rulebook.rule_set_runner - ERROR - Error calling run_playbook
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]: Traceback (most recent call last):
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:   File "/usr/local/lib/python3.11/site-packages/ansible_rulebook/rule_set_runner.py", line 375, in _call_action
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:     await builtin_actions[action](
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:   File "/usr/local/lib/python3.11/site-packages/ansible_rulebook/builtin.py", line 303, in run_playbook
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:     temp_dir, playbook_name = await pre_process_runner(
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:                               ^^^^^^^^^^^^^^^^^^^^^^^^^
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:   File "/usr/local/lib/python3.11/site-packages/ansible_rulebook/builtin.py", line 647, in pre_process_runner
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]:     raise PlaybookNotFoundException(msg)
May 24 12:43:24 ip-172-31-17-143.ec2.internal ansible-rulebook[1164134]: ansible_rulebook.exception.PlaybookNotFoundException: Could not find a playbook for ansible.containerized_installer.install from /home/ec2-user/aap/servicediscovery/etc
```

However, the playbook looks fine, I can run it with ansible-playbook:

```
[ec2-user@ip-172-31-17-143 aap-containerized-installer]$ ansible-playbook ansible.containerized_installer.install
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: running playbook inside collection ansible.containerized_installer
```

It seems the 'playbooks' folder is not searched within the collection.

### Expected results

Action runs playbook using FQCN just fine.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mkanoor**
We were only checking for playbooks in the root directory of the collection. This PR looks for the playbook in the root as well as the playbooks directory

https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html

Fixes #514

Supports .yaml or .yml extensions for rulebooks

https://issues.redhat.com/browse/AAP-9191

### CODE REVIEW

File: ansible_rulebook/collection.py [hunk 1]
```diff
@@ -158,8 +161,8 @@ def find_source_filter(collection, source_filter):
 
 
 def has_playbook(collection, source_filter):
```
Comment (ttuffin) [inline ansible_rulebook/collection.py:163]: Can we fix this wrongly named `source_filter` param at the same time?

File: ansible_rulebook/collection.py [hunk 2]
```diff
@@ -158,8 +161,8 @@ def find_source_filter(collection, source_filter):
...
-    return has_object(collection, source_filter, "", ".yml")
+    return has_object(collection, source_filter, EDA_PLAYBOOKS_PATHS, ".yml")
 
 
 def find_playbook(collection, source_filter):
```
Comment (ttuffin) [inline ansible_rulebook/collection.py:167]: Same here

**Comment by Alex-Izquierdo** [comment]
I wonder if it is feasible to kill two birds with one stone and fix this other bug as well. https://issues.redhat.com/browse/AAP-9191

**Comment by rcarrillocruz** [comment]
path looks good to me, other than the suggestions by @ttuffin and @Alex-Izquierdo , which makes sense to me to address in the patch as they are small in scope.

**Comment by mkanoor** [comment]
The CI error should be fixed by #518

**Review by ttuffin — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]
lgtm

**Pull request merged**